aws/resource_aws_kinesis_stream.go

```diff
@@ -85,36 +92,54 @@ func resourceAwsKinesisStreamCreate(d *schema.ResourceData, meta interface{}) er
 		StreamName: aws.String(sn),
 	}
 
-	_, err := conn.CreateStream(createOpts)
-	if err != nil {
-		if awsErr, ok := err.(awserr.Error); ok {
-			return fmt.Errorf("[WARN] Error creating Kinesis Stream: \"%s\", code: \"%s\"", awsErr.Message(), awsErr.Code())
-		}
-		return err
-	}
+	attemptCount := 1
+	for attemptCount <= KINESIS_MAX_ATTEMPTS {
+		_, err := conn.CreateStream(createOpts)
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				switch code := awsErr.Code(); code {
+				case "LimitExceededException":
+					if strings.Contains(awsErr.Message(), "simultaneously be in CREATING or DELETING") {
+						log.Printf("[DEBUG] Attempt %d/%d: Limit on concurrent kinesis stream creations, sleeping for a bit to throttle back create request", attemptCount, KINESIS_MAX_ATTEMPTS)
+						time.Sleep(KINESIS_LIMIT_EXCEEDED_SLEEP)
+						attemptCount += 1
+					}
+				default:
+					return fmt.Errorf("[WARN] Error creating Kinesis Stream: \"%s\", code: \"%s\"", awsErr.Message(), awsErr.Code())
+				}
+			} else {
+				// Non-AWS exception occurred, give up
+				return fmt.Errorf("Error creating Kinesis stream: %s", err)
+			}
+		} else {
```

Comment:
Since we `return` on errors above I don't think the else block is necessary in this context. It can be just below any retry logic, just assuming we'll only ever get to it if creation was successful -> less nesting => easier to read 🙂